### PR TITLE
Obsolete and migrate legacy wire

### LINKED ISSUE #75840: Mild steel wire and legacy wire compatibility
### Is your feature request related to a problem? Please describe.

Ever since #72285 made fences drop `lc_wire` instead of `wire`, obtaining bulk amounts of `wire` for crafting anything but chainmail armor has become exceedingly difficult, even though (from what I've read) most commercial steel wire *is supposed to be* mild steel wire anyways. There is a need to either unify or migrate more instances of `wire` into `lc_wire` for parity.

### Solution you would like.

Note for myself after the discussion.
Bulk replace all `wire` to `lc_wire`.
Add migration of `wire` to `lc_wire`. (test this with a save before and after).

### Describe alternatives you have considered.

- Making an "any_wire" requirement group containing `wire` and `lc_wire`, making recipes use it and calling it a day

- Bulk overwriting all instances of `wire` to `lc_wire` and dealing with the consequences later

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by XygenSS**
#### Summary
Balance "Obsolete and migrate legacy wire in favor of mild steel wire"

#### Purpose of change
Closes #75840. Legacy steel should be discontinued and split instances of `wire` and `lc_wire` should be unified.

#### Describe the solution
- Migration and item handling
Mass replace `"wire"` with `"lc_wire"`.
Create a migration for `wire` to `lc_wire`.
Create a migration for piano wire to `qt_wire`.
Create an `"any_wire"` requirement group that contains all wires (except legacy wire). Include it in appropriate places.

- Constructions
All constructions and deconstructions of terrain and furniture will use `lc_wire` only, except:
Light, heavy, old, electronic machinery may drop all grades of wire
Reinforced glass drops medium steel wire

- Loot
Itemgroups such as steel mill, farm supplies, hardware stores, exodii metal supplies may contain all grades of wire

- Items, recipes, uncrafts
Most recipes where wire grade does not matter uses `any_wire` requirement group
wire baskets and cages use lc_wire only
remove the recipe that converts wire to scrap
remove the recipe that makes wire from scrap (all wire is now made from matching graded steel lump)
mattresses no longer drop springs, instead drop high steel wire

- chainmail
Create a migration for `chain_link` to `lc_chain_link`
Create a migration for `link_sheet` to `lc_link_sheet`
All recipes that use `chain_link` or `link_sheet` now use appropriate graded steel versions

#### Describe alternatives you've considered
Not using an `any_wire` requirement group, or using it less

#### Testing
Create a world without changes. Spawn items. Update to include changes. Verify proper migration and handling.
- [x] Legacy wire, piano wire, chain link, and chainmail sheet migrates properly
- [x] Cannot craft legacy wire and legacy chainlink/mail anymore
- [x] Deconstruct/bash piano, chain link fence, heavy machinery, reinforced glass, and conveyor
- [x] Test affected itemgroups in debug
- [x] Craft and then disassemble a bra
- [x] Verify crafting wire, chain link, and link sheet in faction camp
#### Additional context
You know, it would be best if we just got rid of medium steel altogether.
Mild steel > case hardened steel
High steel > tempered steel
Pretty elegant I think. Medium steel makes the existence of case hardened and high steels complicated.
But that's an issue for another... issue.

### CODE REVIEW

**Comment by PatrikLundell** [comment]
Testing: As well as you can. Some ideas:
- Make/find save with legacy wire available. Verify it transforms without issue when loaded with new code.
- Examine a number or recipes that used to use legacy wire and verify they now require mild steel wire.
- Craft a couple of random modified recipes to verify they still work but uses mild steel wire.
- Debug spawn some mild steel wire and verify it stacks with migrated wire (so the migrated one doesn't become some weird shadow version).

**Comment by XygenSS** [comment]
Should I keep the `scrap` > `lc_wire` recipe (old recipe was `scrap` > `wire`), or remove it in favor of `lc_steel_chunk` > `lc_wire`?

**Comment by XygenSS** [comment]
How broadly should the "any_wire" requirement group be used? Currently they're used in every recipe involving wire where the wire grade does not matter (anything like space heaters, wooden wheel mounts, vehicle controls, butchering racks...). However mild steel wire is indeed easier to work and weld, and the material properties of different wire grades are different. So the alternative is to only allow mild steel wire, which makes it very commonly used, but makes other wire grades harder to use, which may be a non-issue.

File: data/json/items/generic.json [hunk 1]
```diff
@@ -1043,27 +1043,31 @@
...
     "name": { "str": "hardened steel wire" },
+    "description": "A length of thin, relatively stiff steel wire, made of hardened steel.",
     "material": [ "ch_steel" ]
   },
   {
     "type": "GENERIC",
     "id": "qt_wire",
     "copy-from": "lc_wire",
     "name": { "str": "tempered steel wire" },
+    "description": "A length of flexible and durable steel wire, made of tempered spring steel.  Often used in industrial and automotive springs, and as music wire.",
```
Comment (Maleclypse) [inline data/json/items/generic.json:1070]:
```suggestion
    "description": "A length of flexible and durable steel wire, made of tempered spring steel.  Often used in industrial and automotive springs, and the wire in musical instruments.",
```

**Comment by MikasaTanikawa** [comment]
Should it migrate in-progress items or such migrations are not implemented?

**Comment by XygenSS** [comment]
> Should it migrate in-progress items or such migrations are not implemented?

I'm not aware of any ways of doing that, no.

**Pull request merged**